### DISCUSSION TITLE
How can I configure my errors when they are a 422 regarding an argument on a FASTAPI function?

### DISCUSSION THREAD

**Discussion by RafaelMoreira1180778** [pydantic/pydantic#5237] [Question]
I have the following FASTAPI function to perform a patch:

```python
@app.patch("/foo/bar")
def foo(
    model: MyModel
):
    foo = bar()
```

I have constrained field types for the MyModel class and I am receiving an instance of this class as as argument of this function. I cannot try-catch the 422 HTTP_EXCEPTION that is thrown in case my argument's fields do not respect the constrained types (for instance a regex for a string).

I am trying to catch it using [this StackOverflow answer](https://stackoverflow.com/a/72213866/11432272) and it does work, but this leaves me with a single HTTP 400 error, and if other errors are thrown I cannot handle them correctly. Also, I would like to add some logic to my error handling.

How may I handle these validation errors (HTTP 422: Unprocessable Entity?)

Thank you very much!

**Comment by samuelcolvin** [comment]
1. This is a fastapi question
2. There's not enough information here to be able to help - e.g. what `MyModel` is, what you want to do in the case that the input data doesn't match the model
3. FastAPI has great documentation on custom error handling.

**Reply by RafaelMoreira1180778** [reply]
1. Thank you very much, I will move the discussion over there!
2. My model simply has this:
```python
class MyModel(BaseModel):
    model_name: constr(to_lower=True, regex=r'foo-bar')
    list_1: conlist(constr(regex=r'foobar'), min_items=1)
    list_2: Optional[List[constr(regex=r'foobar')]]
    list_3: Optional[List[constr(regex=r'foobar')]]
```
The users will pass it to the API using an JSON, I would like to throw custom errors whenever the User inputs something that does not follow these rules.
3. I am currently using the Validation error, but I am not sure if this validation error ir only thrown whenever there are prolems on data inputs.
